core/src/main/scala/com/gu/tableversions/core/TableVersions.scala

```diff
@@ -60,4 +118,47 @@ object TableVersions {
     final case class RemovePartition(partition: Partition) extends TableOperation
   }
 
+  /** The current state of the table, including all updates and the current version. */
+  case class TableState(currentVersion: CommitId, updates: List[TableUpdate])
```

Comment:
Would it be safer/clearer to use a data structure a bit more contraining than `List`, like a `SortedSet` or something to guarantee updates order ? The companion object to `TableSate` could still take a `List` for construction yet `updates` would be a sorted thing.

Reply:
Not sure I understand - `List` guarantees the order?

Reply:
Oh you mean, use a `Set` to avoid duplicates but use `SortedSet` to still retain order?

Maybe - I have a feeling that this `List` may become an `fs2.Stream` at some point though, as you often don't care about the full history of the table, just the most recent changes. (See @tomwadeson 's comment linked elsewhere). 

But on that note - the `TableState` type isn't part of the public interface for the version repository, so I should make it protected. At which point, presumably the question of the exact type of this is slightly less important?

Reply:
or as discussed IRL, a comment sayig it is to the discretion of concreate implementation to sort the updates most recent first or smtg

Reply:
Yes I think along with adding an explicit test for this in `TableVersionsSpec` would be useful.